agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFile.java

```diff
@@ -109,6 +116,8 @@
 
     logDir = initLogDir();
     directory = logDir + STATUS_FILE_DIRECTORY;
+    File dir = new File(directory);
+    isReadOnly = dir.canRead() && !dir.canWrite();
```

Comment:
I think we need to mkdirs here before testing `canRead` and `canWrite` (and then we can remove the mkdirs below)

also I think we can handle `canRead = false` better. maybe we should call the flag `notWritable` and make `notWritable = !dir.canWrite()`? otherwise if you can't read we are still trying to write

Reply:
check logDir should work.

Reply:
> check logDir should work.

👍

I think there's still the separate issue when `dir.canRead()` is `false`. In that case, `isReadOnly` is `false` and we try to write.

maybe we should just remove that check? e.g.

```suggestion
    isReadOnly = !dir.canWrite();
```